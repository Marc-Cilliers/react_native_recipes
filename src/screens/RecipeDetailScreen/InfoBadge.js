import React from 'react';
import {StyleSheet, View, Text} from 'react-native';

type Props = {
  icon: string,
  text: string,
};

const InfoBadge = ({icon, text}: Props): Object => {
  return (
    <View style={styles.container}>
      <Text style={styles.icon}>{icon}</Text>
      <Text style={styles.text}>{text}</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    padding: 10,
    borderRadius: 10,
    borderWidth: 1,
  },
  icon: {
    fontFamily: 'FontAwesome5Pro-Light',
    fontSize: 25,
    marginRight: 25,
  },
  text: {
    fontWeight: 700,
  },
});
