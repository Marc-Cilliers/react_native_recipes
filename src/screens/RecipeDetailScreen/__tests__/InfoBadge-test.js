import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import InfoBadge from '../InfoBadge';

it('renders correctly', () => {
  const tree = renderer
    .create(<InfoBadge icon={'\uf823'} text={`4 servings`} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
