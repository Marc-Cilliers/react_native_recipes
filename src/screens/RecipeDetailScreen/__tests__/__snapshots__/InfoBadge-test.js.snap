// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderRadius": 10,
      "borderWidth": 1,
      "flex": 1,
      "flexDirection": "row",
      "padding": 10,
    }
  }
>
  <Text
    style={
      Object {
        "fontFamily": "FontAwesome5Pro-Light",
        "fontSize": 25,
        "marginRight": 40,
        "width": 25,
      }
    }
  >
    ï £
  </Text>
  <Text
    style={
      Object {
        "fontWeight": "700",
      }
    }
  >
    4 servings
  </Text>
</View>
`;
